File: service/src/test/java/by/rekuts/travelagency/service/mocktests/UserServiceImplTest.java
package by.rekuts.travelagency.service.mocktests;

import by.rekuts.travelagency.domain.User;
import by.rekuts.travelagency.repository.UserSpecification;
import by.rekuts.travelagency.service.impl.UserServiceImpl;
import org.junit.Before;
import org.junit.Test;

import java.util.List;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotEquals;
import static org.mockito.Matchers.any;
import static org.mockito.Matchers.anyInt;
import static org.mockito.Mockito.*;

public class UserServiceImplTest {
    private static List<User> users;

    @Before
    public void initializeUsersList() {
        users = List.of(
                new User(1, "superlogin1", "thisispassword"),
                new User(2, "superlogin2", "thisispassword"),
                new User(3, "superlogin3", "thisispassword"));
    }

    @Test
    public void insertTest(){
        var userService = mock(UserServiceImpl.class);
        doNothing().when(userService).insert(any(User.class));
        userService.insert(new User());
        userService.insert(new User());
        verify(userService, times(2)).insert(any());
    }

    @Test
    public void deleteTest() {
        var userService = mock(UserServiceImpl.class);
        doNothing().when(userService).insert(any(User.class));
        userService.delete(6);
        userService.delete(8);
        verify(userService, times(2)).delete(anyInt());
    }

    @Test
    public void getUserByIdTestTrue() {
        var userService = mock(UserServiceImpl.class);
        var specification = new UserSpecification(2);
        var singletonList = List.of(users.get(1));
        when(userService.getList(specification)).thenReturn(singletonList);
        var user = userService.getList(specification).get(0);
        assertEquals("superlogin2", user.getLogin());
    }

    @Test
    public void getUserByIdTestFalse() {
        var userService = mock(UserServiceImpl.class);
        var specification = new UserSpecification(2);
        var singletonList = List.of(users.get(1));
        when(userService.getList(specification)).thenReturn(singletonList);
        var user = userService.getList(specification).get(0);
        assertNotEquals("superlogin1", user.getLogin());
    }

    @Test
    public void getAllUsersTestTrue() {
        var userService = mock(UserServiceImpl.class);
        var specification = new UserSpecification();
        when(userService.getList(specification)).thenReturn(users);
        var tourList = userService.getList(specification);
        assertEquals(3, tourList.size());
    }

    @Test
    public void getAllUsersTestFalse() {
        var userService = mock(UserServiceImpl.class);
        var specification = new UserSpecification();
        when(userService.getList(specification)).thenReturn(users);
        var tourList = userService.getList(specification);
        assertNotEquals(0, tourList.size());
    }
}

File: service/src/test/scala/by/rekuts/travelagency/service/mocktests/UserServiceImplTest.scala
package by.rekuts.travelagency.service.mocktests

import java.util

import by.rekuts.travelagency.domain.User
import by.rekuts.travelagency.repository.UserSpecification
import by.rekuts.travelagency.service.impl.UserServiceImpl
import org.junit.Assert.{assertEquals, assertNotEquals}
import org.junit.{Before, Test}
import org.mockito.Matchers.{any, anyInt}
import org.mockito.Mockito._

class UserServiceImplTest {
  private var users: util.List[User] = _

  @Before
  def initializeUsersList(): Unit = {
    users = util.List.of(
      new User(1, "superlogin1", "thisispassword"),
      new User(2, "superlogin2", "thisispassword"),
      new User(3, "superlogin3", "thisispassword"))
  }

  @Test
  def insertTest(): Unit = {
    val userService = mock(classOf[UserServiceImpl])
    doNothing.when(userService).insert(any(classOf[User]))
    userService.insert(new User)
    userService.insert(new User)
    verify(userService, times(2)).insert(any(classOf[User]))
  }

  @Test
  def deleteTest(): Unit = {
    val userService = mock(classOf[UserServiceImpl])
    doNothing.when(userService).insert(any(classOf[User]))
    userService.delete(6)
    userService.delete(8)
    verify(userService, times(2)).delete(anyInt)
  }

  @Test
  def getUserByIdTestTrue(): Unit = {
    val userService = mock(classOf[UserServiceImpl])
    val specification = new UserSpecification(2)
    val singletonList = util.Arrays.asList(users.get(1))
    when(userService.getList(specification)).thenReturn(singletonList)
    val user = userService.getList(specification).get(0)
    assertEquals("superlogin2", user.login)
  }

  @Test
  def getUserByIdTestFalse(): Unit = {
    val userService = mock(classOf[UserServiceImpl])
    val specification = new UserSpecification(2)
    val singletonList = util.Arrays.asList(users.get(1))
    when(userService.getList(specification)).thenReturn(singletonList)
    val user = userService.getList(specification).get(0)
    assertNotEquals("superlogin1", user.login)
  }

  @Test
  def getAllUsersTestTrue(): Unit = {
    val userService = mock(classOf[UserServiceImpl])
    val specification = new UserSpecification
    when(userService.getList(specification)).thenReturn(users)
    val tourList = userService.getList(specification)
    assertEquals(3, tourList.size)
  }

  @Test
  def getAllUsersTestFalse(): Unit = {
    val userService = mock(classOf[UserServiceImpl])
    val specification = new UserSpecification
    when(userService.getList(specification)).thenReturn(users)
    val tourList = userService.getList(specification)
    assertNotEquals(0, tourList.size)
  }
}
